src/Microsoft.Azure.SignalR/EndpointRouters/DefaultEndpointRouter.cs

```diff
@@ -45,5 +46,38 @@ private ServiceEndpoint[] GetNegotiateEndpoints(IEnumerable<ServiceEndpoint> end
 
             return secondary;
         }
+
+        private ServiceEndpoint GetEndpointAccordingToWeight(ServiceEndpoint[] availableEndpoints)
+        {
+            //first check if weight is available or necessary
+            if (availableEndpoints.Any(endpoint => endpoint.EndpointMetrics.ConnectionCapacity == 0) ||
+                availableEndpoints.Length == 1)
+            {
+                return GetEndpointRandomly(availableEndpoints);
+            }
+
+            var we = new int[availableEndpoints.Length];
+            var totalCapacity = 0;
+            for (var i = 0; i < availableEndpoints.Length; i++)
+            {
+                var endpointMetrics = availableEndpoints[i].EndpointMetrics;
+                var remain = endpointMetrics.ConnectionCapacity -
+                             (endpointMetrics.ClientConnectionCount +
+                              endpointMetrics.ServerConnectionCount);
+                var weight = remain > 0 ? remain : 1;
```

Comment:
if there is a instance with no remain, we should not route clients to that instance.

Reply:
Keep it 1 for simplicity

Reply:
Exclude `remain<=0` ones from `GetNegotiateEndpoints(endpoints)` method? So they won't be in the list of available endpoints as they're overload.

Reply:
if remain=1 and others have capacity like 10k, then this instance will have very low chance of being connected.
Completely removing that will be a breaking change as in fact we could use 10% more quota. 
Also if all instances have used up the quota, they will all have weight 1. The traffic will be equally distributed rather than throwing an exception. I try to keep the behavious same as before.